src/metrics/management/commands/process_crossref_events.py

```diff
@@ -11,62 +11,105 @@
+                                }
                             )
-                        except IntegrityError:
-                            print('Duplicate found, skipping.')
-                    except ident_models.Identifier.DoesNotExist:
+                            if c:
+                                print("ALM created {}".format(event['subj']['pid']))
+                            else:
+                                print("Duplicate {}".format(event['subj']['pid']))
+                    except (ident_models.Identifier.DoesNotExist, ident_models.Identifier.MultipleObjectsReturned):
                         # This doi isn't found
                         pass  # print('{0} not found.'.format(doi))
 
 
-def fetch_crossref_data():
+def setup_prefixes(journals):
     journal_prefixes = list()
-    temp_folder = os.path.join(settings.BASE_DIR, 'files', 'temp')
-
-    for journal in journal_models.Journal.objects.all():
+    for journal in journals:
         if journal.use_crossref:
             try:
                 crossref_prefix = journal.get_setting('Identifiers', 'crossref_prefix')
                 journal_prefixes.append(crossref_prefix)
             except IndexError:
                 print('{0} has no DOI Prefix set.'.format(journal))
 
-    journal_prefixes = set(journal_prefixes)
+    return set(journal_prefixes)
 
-    for prefix in journal_prefixes:
-        file_path = os.path.join(temp_folder, '{prefix}.json'.format(prefix=prefix))
 
-        if not os.path.isfile(file_path):
-            r = requests.get(
-                'https://query.eventdata.crossref.org/events?rows=10000&filter=obj-id.prefix:{0}'.format(prefix))
+def request_crossref_data(prefix, cursor):
+    if cursor:
+        response = requests.get(
+            'https://api.eventdata.crossref.org/v1/events?mailto=andy.byers@openlibhums.org&obj-id.prefix={}&cursor={}'.format(prefix, cursor)
+        )
+    else:
+        response = requests.get(
+            'https://api.eventdata.crossref.org/v1/events?mailto=andy.byers@openlibhums.org&obj-id.prefix={}'.format(prefix)
+        )
+    return response
+
+
+def fetch_crossref_data(prefix, cursor=None):
+
+    file_path = os.path.join(
+        CROSSREF_TEMP_DIR, '{prefix}-{date}.json'.format(
+            prefix=prefix,
+            date=timezone.localdate(),
+        )
+    )
+
+    if not os.path.isfile(file_path):
+        responses = []
+        cursor = None
 
-            with open(file_path, 'w') as file:
-                file.write(json.dumps(r.json()))
+        while True:
+            response = request_crossref_data(prefix, cursor)
+            responses.append(response.json())
+            _dict = response.json()
+            cursor = _dict['message'].get('next-cursor', None)
+
+            if not cursor:
+                break
+
+        for r in responses:
+            with open(file_path, 'a') as file:
+                file.write('{}\n'.format(json.dumps(r)))
```

Comment:
Is there a reason to write to disk instead of handling them directly from memory?